### ISSUE TITLE
[L] Backcountry: Look more into MapLibre for frontend

### ISSUE THREAD

**Issue by Dianadec** [bcgov/reserve-rec-public#29]
#### Description:

Need to learn all about what we can do in maplibre so that we will be ready to use it depending on what the design feedback is and what we are wanting the map to include.
This is applicable to public first, then also admin.

#### Acceptance Criteria:

- [ ] Create first-pass reusable front end map component, explore some rudimentary map functionalities - these will likely be needed regardless of what the designs call for:
   - [x] Set min/max zoom and/or pannable map extents on map load (example - ensure you can't zoom out further than necessary or pan away from British Columbia)
   - [ ] Set map centroid or visible bounding box on map init (example - you want the map to initialize already zoomed in to a location of focus)
   - [x] Provide accessible tooling to zoom/pan (like buttons) overlaid in map
   - [ ] Other rudimentary functions that might be necessary regardless of design
- [ ] Documentation of maplibre learnings is compiled somewhere useful for developers and whoever else needs to see it.
- [ ] Maybe have a share and tell session with whole team (devs + designers) - to help inform design/implementation options moving forward.

#### Dependencies

- Blocked by
- Blocking

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
@marklise to break out the last two AC points in another ticket.

**Comment by marklise** [comment]
We have a pretty basic map working. There's not a whole lot to actually document so it might be best to wait for designs to be made and then go from there. Depending on the designs we may or may not need to do much more work on this particular ticket.

**Comment by LindsayMacfarlane** [comment]
Thanks @marklise ! Before moving to done, can you confirm if the last two acceptance criteria were put into a new ticket(s) as per Diana's note above? I did a quick search in our backlog and didn't see anything. 

![Image](https://github.com/user-attachments/assets/1c4a4b0c-cb96-447e-a8cf-4265a50cd8b8)

**Comment by marklise** [comment]
@LindsayMacfarlane 

https://github.com/bcgov/reserve-rec-public/issues/88
https://github.com/bcgov/reserve-rec-public/issues/89

**Comment by LindsayMacfarlane** [comment]
Thank you @marklise - moving ticket to done :)
